### DISCUSSION TITLE
How To work with Typescript and d3-simple-slider

### DISCUSSION THREAD

**Discussion by Emilion** [johnwalley/d3-simple-slider#185] [Q&A] [answered]
Hello, 

I was trying to implement d3-simple-slider with TypeScript but It's very hard, due to the documentation is not related to the latest version e.g. `sliderBottom()` does not have min, max methods. Also I cannot invoke `sliderBottom()` without passing scale function,  which is very hard to define without reverse engineering the library. 

I will appreciate if you update the readme or just to give some hints how to deal with the library along typescript.

Best regards,

**Comment by johnwalley** [comment, marked as answer]
Hi @Emilion, sorry for not replying earlier. I didn't notice that discussions were enabled for this project!

This project is a little unusual as typically I'd write it in Typescript and get the type definitions for free. However, this project tries to copy the style of the official d3 modules. This is mostly vanilla JS so I rely on keeping the Definitely Typed definitions up to date.

I'll take a look at both the documentation and type definitions when I next get the chance.

Thanks for your input.
